tools/github/before_install.sh

```diff
@@ -31,6 +31,12 @@ fi
 
 python -m pip install --upgrade pip
 
+# TODO: delete when scipy, numpy, cython and pywavelets free-threaded wheels are available on PyPi
+FREE_THREADED_BUILD="$(python -c"import sysconfig; print(bool(sysconfig.get_config_var('Py_GIL_DISABLED')))")"
+if [[ $FREE_THREADED_BUILD == "True" ]]; then
+    pip install --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython numpy scipy pywavelets
```

Comment:
We should keep an eye on whether this actually provides pre-releases in our CI for `FREE_THREADED_BUILD=True`. We've had issues with pip updating indirect dependencies despite the upgrade strategy being "only-if-needed".